sparse-array/src/sparse_array.rs

```diff
@@ -0,0 +1,133 @@
+use crate::bitmap::BitmapIndex;
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct SparseArray<V> {
+    index: BitmapIndex,
+    data: Vec<V>,
```

Comment:
this can be reduced to a `Box<[V]>` to save 8 bytes

Reply:
I'm not sure about that. Will we still be able to add and remove elements at arbitrary positions?

Reply:
no, the `Box<[V]>` is for when the array is immutable/frozen. when in the tree, it's useful to reduce the cost of the array by 8 bytes by leaf. But once you want to update it, you can thaw it into a `Vec` with `into_vec`

Reply:
This will change the interface a bit. Instead of `set(&mut self, idx: u8, value: V)` and `remove(&mut self, idx: u8) -> Option<V>` we will have `set(self, idx: u8, value: V) -> Self` and `remove(self, idx: u8) -> (Self, Option<V>)`.

Anyway, I think that's ok for an immutable struct.

Reply:
yes, I think that's fine. if you want to have a mutable interface, I usually like the pattern of having a `Builder` type which use data structure that are more friendly to modification like Vec in this case.